### DISCUSSION TITLE
Problem configuring and/or starting the debugger in MacOS (using v1.19.8)

### DISCUSSION THREAD

**Discussion by aknari** [microsoft/vscode-cpptools#12125] [Q&A]
Hi!
I am trying to start the debugger on a Apple Silicon MacOS. I am trying to use the Apple Swift version 5.1 lldb which is installed from xcode in /usr/bin/lldb, since it is the only possibility I have, if I am not wrong.

Just for placing you in context, I have a bunch of source files which are finely compiled with CMake, and the step I would like to follow next is not starting the debugger for a single source file but for the executable obtained from the compilation process. Therefore, the step I am trying to execute is "Start the debugger for the target selected". Nonetheless, when trying to do so from Visual Studio Code the following happens:

```output
[proc] Executing the command: /opt/homebrew/bin/cmake --build /Users/xxxx/devel/xxxx/build --config Debug --target xxxx --
[build] ninja: no work to do.
[driver] Compilation completed: 00:00:00.088
[build] Compilation ended with output code 0
[proc] Executing the command: /usr/local/bin/lldb --version
[proc] The command: /usr/local/bin/lldb --version ended with code: 127
```

That is, it tries to run /usr/local/bin/lldb when that executable, the debugger, is not found there. I have tried to look for a place where I could set the complete path of the debugger, but without success.

If anyone would think of a possible solution for this I would appreciate a lot your advice. I've tried several issues, browsing around, without success.

Thank you very much in advance.

**Comment by Dark-Schneider-666** [comment]
Hello,
Install this extension instead
[https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb](url)
I did nothing more, using clang++ to compile when asked and LLDB for debugging when asked (at creating the launch.json) and can debug with the Mac M1, it stops at breakpoints with no issues.
